src/libraries/System.Private.Xml/src/System/Xml/Schema/DataTypeImplementation.cs

```diff
@@ -1126,8 +1125,8 @@ internal override RestrictionFlags ValidRestrictionFlags
 
                     values.Add(typedValue);
                 }
-                array = values.ToArray(_itemType.ValueType);
-                Debug.Assert(array.GetType() == ListValueType);
+                array = ToArray(values, _itemType.ListValueType);
+                Debug.Assert(array.GetType() == _itemType.ValueType.MakeArrayType());
```

Comment:
```suggestion
                Debug.Assert(array.GetType() == _itemType.ValueType.MakeArrayType());
```
Delete this; or move it inside `ToArray` local method so that it is covered by the suppression.